components/ImageSlider/ImageSlider.tsx

```diff
@@ -0,0 +1,296 @@
+import React, { useState, useRef } from 'react';
+import {
+  Dialog, DialogPanel, DialogTitle, Transition, TransitionChild,
+} from '@headlessui/react';
+import Slider from 'react-slick';
+import 'slick-carousel/slick/slick.css';
+import { useOnClickOutside } from 'usehooks-ts';
+import { Container } from '@/components/Container';
+import { FlexBox } from '@/components/FlexBox';
+import { HeroIcon } from '@/components/HeroIcon';
+import { NextPrevButton } from '@/components/ImageSlider/NextPrevButton';
+import { Slide } from '@/components/ImageSlider/Slide';
+import { RichText } from '@/components/RichText';
+import { SrOnlyText, Text } from '@/components/Typography';
+import { type MarginType } from '@/utilities/datasource';
+import { getProcessedImage } from '@/utilities/getProcessedImage';
+import { getIsSbImagePortrait } from '@/utilities/getIsSbImagePortrait';
+import { SbSliderImageType } from '@/components/Storyblok/Storyblok.types';
+import * as styles from './ImageSlider.styles';
+
+type ImageSliderProps = React.HTMLAttributes<HTMLDivElement> & {
+  slides: SbSliderImageType[];
+  isLightText?: boolean;
+  ariaLabel?: string;
+  showExpandLink?: boolean;
+  marginTop?: MarginType;
+  marginBottom?: MarginType;
+}
+
+export const ImageSlider = ({
+  slides,
+  isLightText,
+  ariaLabel,
+  showExpandLink,
+  marginTop,
+  marginBottom,
+  ...props
+}: ImageSliderProps) => {
+  const [activeSlide, setActiveSlide] = useState(0);
+  const [pagerOffset, setPagerOffset] = useState(0);
+  const pagerWindowRef = useRef<HTMLDivElement>(null);
+  const pagerRef = useRef<HTMLUListElement>(null);
+  const sliderRef = useRef<Slider>(null);
+
+  // Modal states and refs
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const modalSliderRef = useRef<Slider>(null);
+  const modalContentRef = useRef<HTMLDivElement>(null);
+
+  useOnClickOutside(modalContentRef, () => {
+    setIsModalOpen(false);
+  });
+
+  const clickPrev = () => {
+    sliderRef.current?.slickPrev();
+  };
+
+  const clickNext = () => {
+    sliderRef.current?.slickNext();
+  };
+
+  const openModal = () => {
+    setIsModalOpen(true);
+  };
+
+  const closeModal = () => {
+    setIsModalOpen(false);
+    sliderRef.current?.slickGoTo(activeSlide, true);
+  };
+
+  // This moves the thumbnail into view when the active slide changes.
+  const adjustPagerPosition = () => {
+    const windowBox = pagerWindowRef.current?.getBoundingClientRect();
+    const pagerBox = pagerRef.current?.getBoundingClientRect();
+    const activeItem =
```

Comment:
You get the active index item number in the afterChange callback, wouldn't it make sense to use that to get the `nth` item in the pager instead of looking for the CSS class?

Reply:
Admittedly I should've cleaned this up a bit more 😂